tools/clang/lib/SPIRV/DebugTypeVisitor.cpp

```diff
@@ -78,43 +74,39 @@ void DebugTypeVisitor::addDebugTypeForMemberVariables(
       continue;
     }
 
-    uint32_t offsetInBits = kUnknownBitSize;
-    if (!unknownPhysicalLayout && field.offset.hasValue())
-      offsetInBits = *field.offset * 8;
-    else
-      unknownPhysicalLayout = true;
-
-    uint32_t sizeInBits = kUnknownBitSize;
-    if (!unknownPhysicalLayout && field.sizeInBytes.hasValue())
-      sizeInBits = *field.sizeInBytes * 8;
-    else
-      unknownPhysicalLayout = true;
+    // Lower this member's debug type.
+    auto *memberDebugType = lowerToDebugType(field.type);
 
     // TODO: We are currently in the discussion about how to handle
     // a variable type with unknown physical layout. Add proper flags
     // or operations for variables with the unknown physical layout.
     // For example, we do not have physical layout for a local variable.
 
+    // Get offset (in bits) of this member within the composite.
+    uint32_t offsetInBits = field.offset.hasValue()
+                                ? offsetInBits = *field.offset * 8
```

Comment:
what is the type of `field` ? is it a pointer? I'm trying to understand `*field`

Reply:
It's an instance of `StructType::FieldInfo` class. `offset` is from `llvm::Optional<uint32_t> offset;`.
`*field.offset` must be interpreted as `*(field.offset)`.

Reply:
Ah got it.